velox/functions/sparksql/tests/MapTest.cpp

```diff
@@ -87,23 +90,150 @@ TEST_F(MapTest, Wide) {
       mapVector);
 }
 
-TEST_F(MapTest, ErrorCases) {
+TEST_F(MapTest, errorCases) {
+  auto inputVectorInt64 = makeNullableFlatVector<int64_t>({1, 2, 3});
+  auto inputVectorDouble = makeNullableFlatVector<double>({4.0, 5.0, 6.0});
+  auto inputVectorDouble1 = makeNullableFlatVector<double>({10.0, 20.0, 30.0});
+  auto inputVectorDouble2 = makeNullableFlatVector<double>({4.1, 5.1, 6.1});
+  auto nullInputVector = makeNullableFlatVector<int64_t>({1, std::nullopt, 3});
+
   // Number of args
-  auto inputVector1 = makeNullableFlatVector<int64_t>({1, 2, 3});
-  auto inputVector2 = makeNullableFlatVector<double>({4.0, 5.0, 6.0});
-  auto mapVector =
-      makeMapVector<int64_t, double>({{{1, 4.0}}, {{2, 5.0}}, {{3, 6.0}}});
-  mapSimple("map(c0)", {inputVector1}, true);
   mapSimple(
-      "map(c0, c1, c2)", {inputVector1, inputVector2, inputVector1}, true);
+      "map(c0)",
+      {inputVectorInt64},
+      true,
+      nullptr,
+      "signature is not supported");
```

Comment:
This error message is not descriptive. How would a user understand what's wrong?

Reply:
here is the actual error message: 
Scalar function signature is not supported: map(BIGINT). Supported signatures: (K,V) -> map(K,V), (K,V,K,V) -> map(K,V), (K,V,K,V,K,V) -> map(K,V).

Scalar function signature is not supported: map(BIGINT, DOUBLE, BIGINT). Supported signatures: (K,V) -> map(K,V), (K,V,K,V) -> map(K,V), (K,V,K,V,K,V) -> map(K,V).

Scalar function signature is not supported: map(BIGINT, DOUBLE, DOUBLE, DOUBLE). Supported signatures: (K,V) -> map(K,V), (K,V,K,V) -> map(K,V), (K,V,K,V,K,V) -> map(K,V).

Scalar function signature is not supported: map(DOUBLE, BIGINT, DOUBLE, DOUBLE). Supported signatures: (K,V) -> map(K,V), (K,V,K,V) -> map(K,V), (K,V,K,V,K,V) -> map(K,V).

Looks the check below can be removed.

```
    VELOX_CHECK(
        args.size() >= 2 && args.size() % 2 == 0,
        "Map function must take an even number of arguments");
```

```
    // Check key and value types
    for (auto i = 0; i < mapSize; i++) {
      VELOX_CHECK_EQ(
          args[i * 2]->type(),
          keyType,
          "All the key arguments in Map function must be the same!");
      VELOX_CHECK_EQ(
          args[i * 2 + 1]->type(),
          valueType,
          "All the key arguments in Map function must be the same!");
    }
```

Reply:
@zhli1142015 Aha...

> Supported signatures: (K,V) -> map(K,V), (K,V,K,V) -> map(K,V), (K,V,K,V,K,V) -> map(K,V).

This suggest that we are only allowing up to 6 arguments to map function. Does Spark also allow only 6 or does it allow many more?

Velox today doesn't allow specifying variadic signatures where pairs of arguments repeat, hence, I imagine we just hard-coded 1, 2, 3 pair signatures. CC: @laithsakka 

Velox applications do not always consult the signatures and therefore may construct a Call expression for map function > 6 arguments. In this case we do need to check argument types in the function itself. It would be nice to add a unit test that calls 'map' with > 6 arguments.

CC: @rui-mo

Reply:
Spark allows more parameters. But Gluten has function signature validation, and would make the plan fallback to spark for this case. And from our experience, most map functions may use 2 parameters. So I think now this should be OK.
If Velox can support  variadic signatures where pairs of arguments repea, that should be awesome.

Reply:
Added UT for > 6 parameters.